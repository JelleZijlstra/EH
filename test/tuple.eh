#!/usr/bin/ehi
t = Tuple.new: 1, "true", false
printvar: t
printvar: t.length: ()
printvar: t.isA: Tuple
for t.length: () count i
  printvar: t->i
end

# Because the EH object model isn't quite implemented, this is the only way we can actually see printvar on action on a tuple
Tuple.printvar = func: -> (global.printvar): self
t2 = Tuple.new: false, [], 42
(t2.printvar: ())
