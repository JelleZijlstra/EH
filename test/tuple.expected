%%stdout%%
@object <Tuple> [
]
@int 3
@bool true
@int 1
@string 'true'
@bool false
@object <Tuple> [
]
%%stderr%%
